### PR TITLE
feat: distributed sessions

### PR DESCRIPTION
**Pull request by joepegler**
## PR-Codex overview
This PR increases the size limit for the `core (esm)` bundle, updates dependencies, and refactors some functions and components.

### Detailed summary
- Increased size limit for `core (esm)` bundle to 20 kB
- Updated dependencies and added new ones
- Refactored functions in `src/actions`, `src/hooks`, and `src/stories/components`
- Added new components in `src/stories/components`

> The following files were skipped due to too many changes: `src/stories/components/SendTransaction.tsx`, `src/actions/createSessionWithDistributedKey.ts`, `src/actions/useSession.ts`, `.github/workflows/docs.yml`, `src/actions/useSessionWithDistributedKey.ts`, `src/actions/useBatchSession.ts`, `src/hooks/useSession.ts`, `src/stories/components/UseSessionWithDistributedKey.tsx`, `src/hooks/useBatchSession.ts`, `src/stories/components/CreateSessionWithDistributedKey.tsx`, `src/hooks/useSessionWithDistributedKey.ts`, `src/hooks/useCreateSessionWithDistributedKey.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/actions/useBatchSession.ts [hunk 1]
```diff
@@ -29,18 +29,13 @@ export const useBatchSessionAction = async (
     true
   )
 
-  const batchSessionParams = await getBatchSessionTxParams(
```
Comment (livingrockrises) [inline src/actions/useBatchSession.ts:32]: why there are changes in this file?

Comment (joepegler) [reply src/actions/useBatchSession.ts:32]: Made some minor backwards-compatible improvements to sessions dx that I tapped into here

**Review by livingrockrises — APPROVED** [review]

**Comment by livingrockrises** [comment]
let's wait for sdk package to publish and runs to pass.

**Comment by livingrockrises** [comment]
checks are failing.

**Review by livingrockrises — APPROVED** [review]

**Pull request merged**